src/server/service/kv.rs

```diff
@@ -28,18 +31,410 @@ use kvproto::kvrpcpb::{self, *};
 use kvproto::raft_cmdpb::{CmdType, RaftCmdRequest, RaftRequestHeader, Request as RaftRequest};
 use kvproto::raft_serverpb::*;
 use kvproto::tikvpb::*;
-use prometheus::HistogramTimer;
+use prometheus::{Histogram, HistogramTimer};
 use tikv_util::collections::HashMap;
 use tikv_util::future::{paired_future_callback, AndThenWith};
 use tikv_util::mpsc::batch::{unbounded, BatchReceiver, Sender};
+use tikv_util::timer::GLOBAL_TIMER_HANDLE;
 use tikv_util::worker::Scheduler;
+use tokio_threadpool::{Builder as ThreadPoolBuilder, ThreadPool};
 
 const SCHEDULER_IS_BUSY: &str = "scheduler is busy";
 const GC_WORKER_IS_BUSY: &str = "gc worker is busy";
 
 const GRPC_MSG_MAX_BATCH_SIZE: usize = 128;
 const GRPC_MSG_NOTIFY_SIZE: usize = 8;
 
+const REQUEST_BATCH_LIMITER_SAMPLE_WINDOW: usize = 30;
+const REQUEST_BATCH_LIMITER_LOW_LOAD_RATIO: f32 = 0.3;
+
+struct HistogramObserver {
+    histogram: Histogram,
+    sum: f64,
+    count: u64,
+}
+
+impl HistogramObserver {
```

Comment:
Can you extract these logic to another file? `kv.rs` is already very complicated..